network/transport/v2/handlers.go

```diff
@@ -133,24 +134,30 @@ func (p *protocol) handleTransactionPayloadQuery(ctx context.Context, peer trans
 		WithField(core.LogFieldTransactionRef, hash.FromSlice(msg.TransactionRef)).
 		Trace("Handling TransactionPayloadQuery")
 
+	// get peer's connection, authentication status is checked later on
+	connection := p.connectionList.Get(grpc.ByConnected(), grpc.ByNodeDID(peer.NodeDID), grpc.ByPeerID(peer.ID))
+	if connection == nil {
+		return fmt.Errorf("unable handle payload request, connection not found (peer=%s)", peer)
+	}
+
 	emptyResponse := &Envelope_TransactionPayload{TransactionPayload: &TransactionPayload{TransactionRef: msg.TransactionRef}}
 	tx, err := p.state.GetTransaction(ctx, hash.FromSlice(msg.TransactionRef))
 	if err != nil {
 		if errors.Is(err, dag.ErrTransactionNotFound) {
 			// Transaction not found
-			return p.send(peer, emptyResponse)
+			return connection.Send(p, &Envelope{Message: emptyResponse}, false)
 		}
 		return err
 	}
 	if len(tx.PAL()) > 0 {
 		// Private TX, verify connection
-		if !peer.Authenticated {
+		if !peer.Authenticated || !connection.Peer().Authenticated { // peer == connection.Peer()
```

Comment:
if they're the same, why is the extrea check required?

Reply:
being extra safe. removed in #1940

Reply:
"being extra safe" sounds like unsure of program flow (or skills). But if it's removed in the next PR, ok.